packages/trimerge-sync/src/lib/Commits.ts

```diff
@@ -0,0 +1,21 @@
+import {Commit, isMergeCommit} from '../types';
+
+export type CommitRefs = {
+    ref: string;
+    baseRef?: string,
+    mergeRef?: string,
+    mergeBaseRef?: string,
+}
```

Comment:
Hey Marcello, I did this without splitting the commits (https://github.com/marcello3d/trimerge-sync/pull/57) I think it would totally work but I do think it's slightly better with them as split just for the explicitness re: we get to express that baseRefs are required for merge commits for one. I also don't find the refs(commit) paradigm that bad.